### PR TITLE
Upgrade wct/polymer test tooling to the latest being used by BrightspaceUI/core

### PR DESCRIPTION
**Pull request by mdgbayly**
Updates versions of sinon, chai, mocha and wct-mocha.
Fix deprecated usages of sinon.stub
Updated eslint-config-brightspace to bring in the lit-element configs

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
   },
   "rules": {
     "no-var": "error",
-    "prefer-const": "error"
+    "prefer-const": "error",
+    "sort-class-members/sort-class-members": "off"
```
Comment (m6jones) [inline .eslintrc.json]: This should be turned for the lit elements!

Comment (mdgbayly) [reply .eslintrc.json]: Yeh - i have not setup the PR yet, but I have a nested .eslintrc for the lit element based components I am writing for the activity-editor stuff, which extends the brightspace/lit-config which seems to ensure this override is ignored for the lit-elements. 

But just adding the relevant plugins and updating eslint-config-brightspace caused lots of errors to be generated in existing Polymer 3 classes, because the latest version of the polymer-3-config also configures sort-class-members even though that hasn't been enforced in all the existing polymer 3 classes. So adding this override here worked around that. At some point it would be nice for someone to go in an clean up the existing polymer 3 stuff so it conforms.

**Comment by mdgbayly** [comment]
@m6jones Hi Maya

Would you be able to take a look at this now - i think the Travis/Sauce timeouts that was causing CI jobs to fail are now resolved (worked around!)

After much back and forth, I was final able to get it passing tests consistently. The problem turned out to be related to a 60s timeout in wct-mocha relating to loading test files. I'm not sure why we were not getting it in the old version, but maybe something in the newer tooling is loading more files.  I was able to bump this timeout in `test/index.html' prior to the test runner loading the test files.

I almost gave up on trying to update to the latest versions of the polymer testing tools due to the timeouts but it still seems beneficial to update to the latest test tools being used by BrightspaceUI/core too.  I find there are issues running individual tests directly in the browser with the old versions of `web-component-tester` and `browser.js` due to some dependencies on lodash 3 which is deprecated and causes security warnings if added to the project.

File: .eslintrc.json (hunk 1 above)
Comment (m6jones) [reply .eslintrc.json]: It was a late add and these components were made before.

File: .travis.yml [hunk 2]
```diff
@@ -31,4 +31,4 @@ env:
...
\ No newline at end of file
+  - secure: QP5YZ7oT1NAyyLDhNG/kHii+SasV5zR/9XO5ePl2ThAevdpEgrLLk1A76NV0XZMMSRIdvgzMZIvPqHWfA81I2O7QCzXOJKQ+xj8hanGqUUNlGxgDMo8FGmqf7OWyxw+XHoM5E0P20yntmDQnCsYnuPvTwYyZ/6ujQgTbPli2ewvoZL+Doh+RnOHmc7R8HbdpU2Cv42PIZr/H/nW6tpuliP1fNrMvKwzjbRu+zh/ag62CkPOk0jbu6klXTwdwytb0jseY7Y/ybJ/0kOZGmgCguH3Lb2kj4Uncp78aQiAZFyCIurb6iXVZ5K7YtABbpWj2sRtMkJwQedTSCUY7PI8FTuvNowZ9URmJ5d4FAAbo03D6jgur4SHxakyV+iQj5z0vfmw5bJT/ykJzXFgyRTkjy/zznZE3arDeyRiLPKGmlKSYDS51eXUhJ2yqgFtsw3ArCqIoIm3xxe9Qy2DNsJllf/SGIhcGhPXIlG4Mj/59OiqNVJuXORJkwuj4SNK4ujkTGkHWYgwGEglhF9nCSPBOrITaFtyfECNvHc9HM/SeYyaDjMqrvXnGusoU6BKhxqiwhb2LhqrK9SRq6ibeP9ykjvi3PkrmbyVHKTAzXHxKJ5uS1orwQ7oOnC4p751pjZB570wIg4ENUsPML/EwbtfsqWjCYnzGZqOgP8mYDXDprr8=
```
Comment (m6jones) [inline .travis.yml]: Why did you need to change the key?

File: test/d2l-activity-list-item/d2l-activity-list-item-test.js [hunk 3]
```diff
@@ -115,6 +115,10 @@ describe('d2l-activity-list-item', () => {
 				afterNextRender(component, done);
 			});
 
+			afterEach(() => {
```
[the remark this answers is not in the document]
Comment (m6jones) [reply test/d2l-activity-list-item/d2l-activity-list-item-test.js]: Cool, I didn't know that :)
